### DISCUSSION TITLE
How to develop a code generator?

### DISCUSSION THREAD

**Discussion by olmohake** [hasura/graphql-engine#9542] [Question]
We are currently trying to setup a custom code generator, but struggling to point hasura to our generator file. 
I was expecting to be able to do this via _HASURA_GRAPHQL_ACTION_CODEGEN_URI_ or _actions.codegen.uri_  however this setting seems to have no effect, as hasura keeps telling me the following:
```shell
could not find codegen config. For adding codegen config, run:  hasura actions use-codegen 
```
Running  _hasura actions use-codegen_ only enables me to setup an existing code generator, which is not what i am looking for. Is there anyway to point hasura to a local code generator file for development?

**Comment by meetzaveri** [comment]
Hello @OLM0 

From what I am reading, you need to build a custom codegen which isn't in lists provided by hasura CLI. Therefore, you'll need to contribute to codegen assets to build one. Currently, there isn't a way to point to local codegen. 

Here's the link to contribution guide - https://github.com/hasura/codegen-assets/blob/master/builder-kit/README.md

Once you've built a custom codegen, file a PR against this repository - https://github.com/hasura/codegen-assets. We'll take it forward then.

Does that help your question ?
